File: 896-monotonic-array/896-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        int incr = 0;
        for(int i = 1; i < nums.length; i++){
            if(nums[i] > nums[i - 1]){
                if(incr == 0)
                    incr = 1;
                else if (incr != 1)
                    return false;  
            } else if (nums[i] < nums[i - 1]) {
                if(incr == 0)
                    incr = -1;
                else if (incr != -1)
                    return false;   
            }
        }
        return true;
    }
}

File: 896-monotonic-array/896-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        increasing = 0
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                continue
                
            if nums[i] > nums[i - 1]:
                if increasing == 0:
                    increasing = 1
                elif increasing != 1:
                    return False
            else:
                if increasing == 0:
                    increasing = -1
                elif increasing != -1:
                    return False
        return True
        